gapis/resolve/memory.go

```diff
@@ -238,7 +229,17 @@ func valAsString(values []*memory_box.Value) (*pod.Value, []*memory_box.Value) {
 		Val: &pod.Value_String_{
 			String_: sb.String(),
 		},
-	}, values
+	}
+}
+
+// charsFromValues converts the int values in the array to a char array
+func charsFromValues(values []*memory_box.Value) []*memory_box.Value {
```

Comment:
I think that this still modifies the argument.

I want to make sure that I understand correctly. 

This function takes a pointer to an array of `value` struct and returns an array of `value` struct (not a `char` array for some reason).

But while doing it it also modifies the argument and returns it. Is that correct? 

I am afraid that it's not a very good naming in this case because, if I would call a `charsFromValues`, I would expect to get a `char` array from values and values to stay **unmodified**.

I don't know why we need the value struct so I cannot say anything about it.

Assuming that the `value` struct is needed, maybe we should rename this function as `convertValueTypeToChar` or something similar that does return void(or it can still return an array of `value` struct, if we are doing some function chaining but I cannot see that it's happening).

Reply:
Renamed to convertValueTypeToChar